src/Markdig/Helpers/StringSlice.cs

```diff
@@ -208,19 +242,22 @@ public bool MatchLowercase(string text, int offset = 0)
         /// <param name="end">The end.</param>
         /// <param name="offset">The offset.</param>
         /// <returns><c>true</c> if the text matches; <c>false</c> otherwise</returns>
-        public bool MatchLowercase(string text, int end, int offset)
+        public readonly bool MatchLowercase(string text, int end, int offset)
         {
             var index = Start + offset;
-            int i = 0;
-            for (; index <= end && i < text.Length; i++, index++)
+
+            if (end - index + 1 < text.Length)
+                return false;
+
+            string sliceText = Text;
```

Comment:
All the work you have done in this PR to move field access to single local variable is indeed helping a lot the codegen, amazing job